### ISSUE TITLE
[a11y]: Provide clearer labellling for Codesnippet copy buttons

### ISSUE THREAD

**Issue by dakahn** [carbon-design-system/carbon#10526]
Two accessibility improvements were suggested for Codesnippet in the Carbon Accessibility Guild:

- `<button>` elements should be aria labelled to describe their functionality. for example "Copy to clipboard". These buttons can be `aria-describedby` the copyable text itself
- inline, single line, and multiline copy buttons should show tooltips on hover/focus

**Comment by mbgower** [comment]
there are some considerations for the accessible name needing to include the text of the button. So inline code snippets may need a bit more consideration, as aria-describedby does not become part of accessible name

**Comment by guidari** [comment]
Hey @dakahn 
Looks like your issue have already been adressed by the team. Also in the code the `aria-labels` are there.

Is tehre anything I am missing here? Otherwise we can close this issue.

![Screenshot 2024-01-09 at 17 19 26](https://github.com/carbon-design-system/carbon/assets/52183462/1513a719-a25c-4b1d-9aa8-ba026189cf88)
